instrumentation/akka/akka-http-10.0/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/akkahttp/AkkaHttpServerRouteTest.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.akkahttp
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.Directives.{
+  IntNumber,
+  complete,
+  concat,
+  path,
+  pathEndOrSingleSlash,
+  pathPrefix,
+  pathSingleSlash
+}
+import akka.http.scaladsl.server.Route
+import akka.stream.ActorMaterializer
+import io.opentelemetry.instrumentation.test.utils.PortUtils
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension
+import io.opentelemetry.sdk.testing.assertj.{SpanDataAssert, TraceAssert}
+import io.opentelemetry.testing.internal.armeria.client.WebClient
+import io.opentelemetry.testing.internal.armeria.common.{
+  AggregatedHttpRequest,
+  HttpMethod
+}
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{AfterAll, Test, TestInstance}
+import org.junit.jupiter.api.extension.RegisterExtension
+
+import java.net.{URI, URISyntaxException}
+import java.util.function.Consumer
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.Await
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
Perhaps add a comment why the test instance is per class.

Reply:
Here I think it is because of the `@AfterAll` annotated cleanup method that would otherwise need to be static. We use `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` in multiple tests and none of them have a comment why it is used, but I'm sure that removing that annotation for most of them will make tests fail. I personally think that this comment wouldn't be too useful.